diff --git a/scripts/cross-build.sh b/scripts/cross-build.sh
index 0e98c0c..8e369bd 100755
--- a/scripts/cross-build.sh
+++ b/scripts/cross-build.sh
@@ -108,2 +108,15 @@ build_libwebsockets() {
         sed -i 's/ websockets_shared//g' cmake/libwebsockets-config.cmake.in
+patch -u lib/roles/ws/ext/extension-permessage-deflate.c <<'EOF'
+diff --git a/lib/roles/ws/ext/extension-permessage-deflate.c b/lib/roles/ws/ext/extension-permessage-deflate.c
+index 790db4e0a..b492e78ce 100644
+--- a/lib/roles/ws/ext/extension-permessage-deflate.c
++++ b/lib/roles/ws/ext/extension-permessage-deflate.c
+@@ -311 +310,0 @@ lws_extension_callback_pm_deflate(struct lws_context *context,
+-		/* COV says we can overflow if "eb_in.len == 0 and rx->avail_in == 4" */
+@@ -313,4 +311,0 @@ lws_extension_callback_pm_deflate(struct lws_context *context,
+-		if ((unsigned int)priv->rx.avail_in > (unsigned int)pmdrx->eb_in.len) {
+-			lwsl_wsi_err(wsi, "rx buffer underflow");
+-			return PMDR_FAILED;
+-		}
+EOF
         rm -rf build && mkdir -p build && cd build
